menuconfig.py

```diff
@@ -165,100 +165,255 @@
+# underline     Use underline text
+# standout      Standout text attribute (reverse color)
+#
+# A word without '=' is considered a style template defition. A style template
+# is used as the basis for the current style. If not style template is defined
+# the default style is ued.
+#
+_STYLES = {
+    "default": """
+    path=fg:black,bg:white,bold
+    separator=fg:black,bg:yellow,bold
+    list=fg:black,bg:white
+    selection=fg:white,bg:blue,bold
+    inv_list=fg:red,bg:white
+    inv_selection=fg:red,bg:white
+    help=fg:black,bg:white,bold
+    frame=fg:black,bg:yellow,bold
+    body=fg:white,bg:black
+    edit=fg:white,bg:blue
+    jump=fg:white,bg:blue
+    text=fg:black,bg:white
+    """,
+
+    # This style is forced on terminals that do no support colors
+    "monochrome": """
+    path=bold
+    separator=bold,standout
+    list=
+    selection=bold,standout
+    inv_list=bold
+    inv_selection=bold,standout
+    help=bold
+    frame=bold,standout
+    body=
+    edit=standout
+    jump=
+    text=
+    """,
+
+    # Blue tinted style loosely representing lxdialog
+    "aquatic": """
+    path=fg:cyan,bg:blue,bold
+    separator=fg:white,bg:cyan,bold
+    help=fg:cyan,bg:blue,bold
+    frame=fg:white,bg:cyan,bold
+    body=fg:white,bg:blue
+    edit=fg:black,bg:white
+    """
+}
+
+_CUSTOM_STYLE = None
+
+# Standard colors definition
+_STYLE_STD_COLORS = {
+    # Basic colors
+    "black":        curses.COLOR_BLACK,
+    "red":          curses.COLOR_RED,
+    "green":        curses.COLOR_GREEN,
+    "yellow":       curses.COLOR_YELLOW,
+    "blue":         curses.COLOR_BLUE,
+    "magenta":      curses.COLOR_MAGENTA,
+    "cyan":         curses.COLOR_CYAN,
+    "white":        curses.COLOR_WHITE,
+
+    # Bright versions
+    "brightblack":  curses.COLOR_BLACK + 8,
+    "brightred":    curses.COLOR_RED + 8,
+    "brightgreen":  curses.COLOR_GREEN + 8,
+    "brightyellow": curses.COLOR_YELLOW + 8,
+    "brightblue":   curses.COLOR_BLUE + 8,
+    "brightmagenta":curses.COLOR_MAGENTA + 8,
+    "brightcyan":   curses.COLOR_CYAN + 8,
+    "brightwhite":  curses.COLOR_WHITE + 8,
+
+    # Aliases
+    "purple":       curses.COLOR_MAGENTA,
+    "brightpurple": curses.COLOR_MAGENTA + 8,
+    "default":  -1
+}
+
+
+class StyleError(Exception):
+    pass
+
+def _parse_style_expand(style_str):
+    style = {}
+
+    for sline in style_str.split():
+        skey = sline.split("=")
+        # Words without a "=" character represents a style template
+        if len(skey) == 1:
+            # Recursively expand style templates
+            if skey[0] not in _STYLES:
+                raise StyleError("Unknown built-in style: {}. Error near: {}".format(
+                    skey[0], sline))
+
+            # Merge in the template
+            tstyle = _parse_style_expand(_STYLES[skey[0]])
+            style.update(tstyle)
+
+        # Key=data value, store it
+        elif len(skey) == 2:
+            style[skey[0]] = skey[1]
```

Comment:
Could move the `_style_to_curses()` call to here, to get rid of the calls in `_init_styles()`.

Since `_style_to_curses()` should already detect an error if `skey[1]` has an extra `=` in it, `_parse_style_expand()` could then be simplified too:

```
for sline in style_str.split():
    if "=" in sline:
        key, val = sline.split("=")
        style[key] = _style_to_curses(val)

    else:
        ...
        style.update(_parse_style_expand(_STYLES[sline]))

Reply:
Fixed.